packages/devtools_app/lib/src/debugger/debugger_model.dart

```diff
@@ -2,14 +2,61 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:meta/meta.dart';
+import 'package:pedantic/pedantic.dart';
 import 'package:vm_service/vm_service.dart';
 
+import '../config_specific/logger/logger.dart';
+import '../globals.dart';
+import '../inspector/diagnostics_node.dart';
+import '../inspector/inspector_service.dart';
 import '../trees.dart';
 import '../ui/search.dart';
 import '../utils.dart';
 
+/// Whether to include properties and children surfaced through
+/// Diagnosticable objects as part of the generic Debugger view of an object.
+bool includeDiagnosticPropertiesInDebugger = true;
+
+/// Set to false as it is hard to avoid confusing overlap between the children
+/// visible under fields for typical objects and we don't have a great way of
+/// clarifying that these are children from the Diagnostic view of the object
+/// which might be different from children on fields for the Inspector summary
+/// tree case which has a filtered view of children.
+bool includeDiagnosticChildren = false;
+
+/// A generic [InstanceRef] using either format used by the [InspectorService]
+/// or Dart VM.
+///
+/// Either one or both of [instanceRef] and [diagnostic] may be provided. The
+/// `valueRef` getter on the [diagnostic] should refer to the same object as
+/// [instanceRef] although using the [InspectorInstanceRef] scheme.
+/// A [RemoteDiagnosticsNode] is used rather than an [InspectorInstanceRef] as
+/// the additional data provided by [RemoteDiagnosticsNode] is helpful to
+/// correctly display the object and [RemoteDiagnosticsNode] includes a
+/// reference to an [InspectorInstanceRef].
+class GenericInstanceRef {
+  GenericInstanceRef({
+    @required this.isolateRef,
+    this.instanceRef,
+    this.diagnostic,
+  });
+
+  final InstanceRef instanceRef;
+
+  /// If both [diagnostic] and [instanceRef] are provided, [diagnostic.valueRef]
+  /// must reference the same underlying object just using the
```

Comment:
should we add an assert for this in the constructor?

Reply:
unfortunately no because the two refs are using different schemes (VMService vs InspectorService) so we couldn't verify they are equivalent with a roundtrip to the device.